### PR TITLE
Add "corestone-core"

### PR DESCRIPTION
**Pull request by iamssen**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
    - I can't pass the `npm test`. Is there a way to test only cornerstone-core?
```
A       types/yarnpkg__lockfile/tslint.json
A       types/yarnpkg__lockfile/yarnpkg__lockfile-tests.ts
D       types/year-days/index.d.ts
D       types/year-days/tsconfig.json
D       types/year-days/year-days-tests.ts
M       types/yeoman-environment/index.d.ts
M       types/yeoman-generator/index.d.ts
M       types/yog-bigpipe/index.d.ts
M       types/youtube/index.d.ts
M       types/yup/index.d.ts
M       types/yup/yup-tests.ts
M       types/z-schema/tsconfig.json
A       types/zchat-browser/index.d.ts
A       types/zchat-browser/tsconfig.json
A       types/zchat-browser/tslint.json
A       types/zchat-browser/zchat-browser-tests.ts
A       types/zdog/index.d.ts
A       types/zdog/tsconfig.json
A       types/zdog/tslint.json
A       types/zdog/zdog-tests.ts
A       types/zenaton/index.d.ts
A       types/zenaton/tsconfig.json
A       types/zenaton/tslint.json
A       types/zenaton/zenaton-tests.tswarning: inexact rename detection was skipped due to too many files.
warning: you may want to set your diff.renameLimit variable to at least 2291 and retry the command.
AssertionError [ERR_ASSERTION] [ERR_ASSERTION]: The specified version 5.1.0 of active-win must be newer than the version
it is supposed to replace, 5.1.0 of @types/active-win.
    at checkNotNeededPackage (/Users/ssen/Workspace/DefinitelyTyped/node_modules/types-publisher/src/tester/test-runner.ts:135:5)
    at getAffectedPackagesFromDiff (/Users/ssen/Workspace/DefinitelyTyped/node_modules/types-publisher/src/tester/test-runner.ts:89:13)
    at Object.runTests [as default] (/Users/ssen/Workspace/DefinitelyTyped/node_modules/types-publisher/src/tester/test-runner.ts:71:65)
    at main (/Users/ssen/Workspace/DefinitelyTyped/node_modules/types-publisher/src/tester/test.ts:35:5) {
  generatedMessage: false,
  code: 'ERR_ASSERTION',
  actual: false,
  expected: true,
  operator: '=='
}
npm ERR! Test failed.  See above for more details.
```
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldnt have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/cornerstone-core/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,1517 @@
...
+/**
+ * Maps scalar values into colors via a lookup table
+ * LookupTable is an object that is used by mapper objects to map scalar values
+ * into rgba (red-green-blue-alpha transparency) color specification, or rgba into scalar values.
+ * The color table can be created by direct insertion of color values,
+ * or by specifying hue, saturation, value, and alpha range and generating a table
+ */
+export interface LookupTable {
+    // TODO
+    __foo__: 'bar';
```
Comment (amcasey) [inline types/cornerstone-core/index.d.ts]: This is being used as a brand?

File: types/cornerstone-core/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,1517 @@
...
+/**
+ * Gets metadata from the registered metadata providers. Will call each one from highest priority to lowest until one responds
+ *
+ * @param type The type of metadata requested from the metadata store
+ * @param imageId The Cornerstone Image Object's imageId
+ * @returns The metadata retrieved from the metadata store
+ *
+ * TODO check clearitify the unknown type
+ */
+export function getMetaData(type: string, imageId: string): unknown;
```
Comment (amcasey) [inline types/cornerstone-core/index.d.ts]: To confirm, the expectation is that the result of `getMetaData` is intended to be difficult to consume without an explicit cast?

**Review by amcasey — CHANGES_REQUESTED** [review]
Providing an incomplete type definition is fine, but I have some questions about the way the incomplete sections were handled.

**Comment by iamssen** [comment]
@amcasey cornerstone-core was a huge library to work alone. I checked the source code and defined it, but it was difficult to check everything.

I temporarily wrote unknowns that required a lot of time to check the source code and then marked TODO.

I also made some garbage code to avoid dts-lint ... (`__foo__: 'bar'` and `__UnknownFunction__`...)

There was too much work to do, so I tried to upload this unfinished definition, but I think I need to close this Pull Request and do more work.

I'm not sure how to work with such large definitions that are difficult to complete at once. Is there a similar definition for reference?

**Comment by amcasey** [comment]
@iamssen, I think it's just fine to submit part of a definition - especially for a very large library.  I'd suggest using `any`, rather than `unknown`, if you're leaving loose ends (which, again, is fine) and suppress the lint warning about empty types, rather than giving them dummy members.

I'm not sure what sort of example you're looking for.  If you provide more details, I'll see what I can find.

**Comment by iamssen** [comment]
@amcasey I was thinking about choosing between "unknown" and "any". Providing these uncertain parts as "unknown" will likely cause users to use them more carefully. Conversely, providing it as "any" may lead users to write wrong code.

Users will write a lot of "Type Guard", but is it better than writing insecure code?

-----

```
interface SomeType {
  __dont_use_this__: 'this is an dummy member for supress lint warning';
  // TODO describe type
}
```

Can I give a dummy member with code like this?

**Comment by iamssen** [comment]
I supplemented the cornerstone-core definition and found that the API documentation differs from the actual API a lot.

I think I will have to rework almost everything, so I close this PR.

**Comment by amcasey** [comment]
Thanks, we're looking forward to the new version!

**Pull request closed without merging**